### PR TITLE
amrex: Add CUDA variant

### PR DESCRIPTION
**Pull request by eschnett**

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]
@mic84 @asalmgren can you review?

File: var/spack/repos/builtin/packages/amrex/package.py [hunk 1]
```diff
@@ -67,10 +67,13 @@ class Amrex(CMakePackage):
             description='Enable Hypre interfaces')
     variant('petsc', default=False,
             description='Enable PETSc interfaces')
+    variant('cuda', default=False,
+            description='Enable CUDA interfaces')
 
     # Build dependencies
     depends_on('mpi', when='+mpi')
     depends_on('sundials@4.0.0:4.1.0 +ARKODE +CVODE', when='@19.08: +sundials')
+    depends_on('cuda', when='+cuda')
```
Comment (mic84) [inline var/spack/repos/builtin/packages/amrex/package.py]:
```suggestion
    depends_on('cuda@9.0.0:', when='+cuda')
```

Comment (mic84) [reply var/spack/repos/builtin/packages/amrex/package.py]: AMReX requires at least CUDA 9.0

Comment (eschnett) [reply var/spack/repos/builtin/packages/amrex/package.py]: Done.

**Review by mic84 — APPROVED** [review]

**Pull request merged**